src/python/strelka/scanners/scan_ics.py

```diff
@@ -0,0 +1,479 @@
+        """Extract file from data URI with base64 content."""
+        try:
+            import base64
+            
+            if ';base64,' not in data_uri:
+                return
+                
+            header, encoded_data = data_uri.split(';base64,', 1)
+            mime_type = header.replace('data:', '')
+            decoded_data = base64.b64decode(encoded_data)
+            
+            filename = (self._get_param_value(attachment_data['params'], ['X-FILENAME', 'FILENAME']) or 
+                       self._generate_filename(mime_type))
+            
+            self._create_extracted_file(decoded_data, filename, mime_type, attachment_data, expire_at)
+            
+        except Exception as e:
+            self.flags.append('data_uri_decode_error')
+            attachment_data['decode_error'] = str(e)
+    
+    def _extract_base64_binary_attachment(self, base64_data, attachment_data, expire_at):
+        """Extract base64-encoded binary attachment (ENCODING=BASE64)."""
+        try:
+            import base64
+            
+            # Decode base64 data
+            decoded_data = base64.b64decode(base64_data)
+            
+            # Get filename and mime type from parameters
+            filename = self._get_param_value(attachment_data['params'], ['X-FILENAME', 'FILENAME'])
+            mime_type = self._get_param_value(attachment_data['params'], 'FMTTYPE')
+            
+            if not filename:
+                filename = f'ics_attachment_{self.event["total"]["attachments"]}'
+            
+            self._create_extracted_file(decoded_data, filename, mime_type, attachment_data, expire_at)
+            
+        except Exception as e:
+            self.flags.append('base64_binary_decode_error')
+            attachment_data['decode_error'] = str(e)
+    
+    def _get_param_value(self, params, param_names):
+        """Get parameter value by name(s)."""
+        if isinstance(param_names, str):
+            param_names = [param_names]
+        
+        for param in params:
+            if param.get('name') in param_names:
+                return param.get('value')
+        return None
+    
+    def _generate_filename(self, mime_type):
+        """Generate filename from MIME type."""
+        ext_map = {
+            'application/pdf': '.pdf',
+            'text/plain': '.txt', 
+            'application/json': '.json',
+            'image/png': '.png',
+            'image/jpeg': '.jpg'
+        }
+        ext = ext_map.get(mime_type, '.bin')
+        return f'ics_data_uri_{self.event["total"]["attachments"]}{ext}'
+    
+    def _create_extracted_file(self, data, filename, mime_type, attachment_data, expire_at):
+        """Create and upload extracted file."""
+        extract_file = strelka.File(name=filename, source=self.name)
+        
+        if mime_type:
+            extract_file.add_flavors({'external': [mime_type]})
+            attachment_data['mime_type'] = mime_type
```

Comment:
we should probably do metadata generation of the attachment outside of the _create_extracted_file, and instead allow _create_extracted_file to just handle  submitting the extracted attachment back into Strelka